javaslang/src/test/java/javaslang/collection/TreeMapTest.java

```diff
@@ -122,6 +126,22 @@ public void shouldWrapMap() {
         assertThat(TreeMap.ofAll(source)).isEqualTo(emptyIntInt().put(1, 2).put(3, 4));
     }
 
+    // -- ofAll
+
+    @Test
+    public void shouldCreateKeyComparatorForJavaUtilMap() {
+        final TreeMap<String, Integer> actual = TreeMap.ofAll(mapOfTuples(Tuple.of("c", 0), Tuple.of("a", 0), Tuple.of("b", 0)).toJavaMap());
+        final List<String> expected = List.of("a", "b", "c");
+        assertThat(actual.keySet().toList()).isEqualTo(expected);
```

Comment:
@paplorinc But wait - what is the result of `actual.toList()`? Because `TreeMap<K, V>` is and ordered Traversable<Tuple2<K,V>>` I would expect that `toList()` returns the contained entries in the underlying order. If that isn't the case it is a bug.